### ISSUE TITLE
Problem using periods in user names

### ISSUE THREAD

**Issue by brianrc** [timersys/facebook-login#9]
The current plugin uses periods to concatenate first and last time, but there is a problem with that and BuddyPress. When the username shows up in the user profile the periods are replaced by hyphens. I think this is because periods aren't allowed in certain areas of the URL and BuddyPress uses the username as part of the URL. 

This leaves a disconnect for the user between their actual username, first.last, with the username they are using on the rest of the site, first-last. Users can't login with the latter, they have to use the former. I think this creates a lot of confusion (especially since this a generated username, not one they picked for themselves).

![bp-user-profile](https://cloud.githubusercontent.com/assets/573939/12206109/1310a884-b60d-11e5-9883-44f9a78e1399.png)

In conclusion, I would propose maybe using hypens (-) or underscores (_) or nothing in between first and last name when generating usernames.

**Comment by timersys** [comment]
Changed to - and use sanitize_title to remove any extra dots . e5d1c89
